dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/WorkFlowLineageMapper.xml

```diff
@@ -161,4 +161,58 @@
         </if>
         AND a.task_type = 'DEPENDENT'
     </select>
+
+    <select id="queryTaskDepOnProcess" resultType="org.apache.dolphinscheduler.dao.entity.TaskMainInfo">
+        select td.id
+            , td.name as taskName
+            , td.code as taskCode
+            , td.version as taskVersion
+            , td.task_type as taskType
+            , ptr.process_definition_code as processDefinitionCode
+            , pd.name as processDefinitionName
+            , pd.version as processDefinitionVersion
+            , pd.release_state as processReleaseState
+        from t_ds_task_definition td
+        join t_ds_process_task_relation ptr on ptr.post_task_code = td.code and td.version = ptr.post_task_version
+        join t_ds_process_definition pd on pd.code = ptr.process_definition_code and pd.version = ptr.process_definition_version
+        <where>
+            <if test="projectCode != 0">
+                and ptr.project_code = #{projectCode}
+            </if>
+            <if test="processDefinitionCode != 0">
+                and ptr.process_definition_code != #{processDefinitionCode}
+                and (
```

Comment:
The "definitionCode" of this method should be changed to "processDefinitionCode" under subprocess.

Reply:
do you mean in case of task subprocess, we should use ('%"processDefinitionCode":', #{processDefinitionCode}, '%')`？
we here will check both task type `subprocess` and task type `dependent`.

* for subprocess: using `concat('%"processDefinitionCode":', #{processDefinitionCode}, '%')` in L186
* for dependent: using `concat('%"definitionCode":', #{processDefinitionCode}, '%')` in L185

Reply:
maybe I should add some comment about it, thank for you metion

Reply:
sorry, I saw lines 185 and 186 and the relationship is or.

Reply:
I add some comment in 98264fa to make it more specific